### PR TITLE
planner: refactor code of Global Binding

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #51347

Problem Summary: planner: refactor code of Global Binding

### What changed and how does it work?

Merge CrossDBBindingCache into BindingCache to simplify the code, no logical change, only refactor.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/bindinfo/binding_cache.go [hunk 1]
```diff
@@ -448,11 +390,28 @@ func (c *bindingCache) GetAllBindings() Bindings {
...
+		noDBDigests = append(noDBDigests, noDBDigest)
+	}
+
+	for i, binding := range bindings {
+		c.digestBiMap.Add(noDBDigests[i], binding.SQLDigest)
+	}
+
 	cacheKey := bindingCacheKey(sqlDigest)
-	_, err = c.set(cacheKey, meta)
+	_, err = c.set(cacheKey, bindings)
```
Comment (Rustin170506) [inline pkg/bindinfo/binding_cache.go:414]: It seems we dropped the original comment here.

**Review by Rustin170506 — APPROVED** [review]
Thanks! :shipit:

/hold

Not sure if we need to add the comment back.

File: pkg/bindinfo/binding_cache.go (hunk 1 above)
Comment (qw4990) [reply pkg/bindinfo/binding_cache.go:416]: Updated. Thanks.

**Pull request merged**